### ISSUE TITLE
Expand manual

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#1985]
There are some topics that emerged repeatedly over the last few month (and especially during Christmas break). Given the frequency, these topics deserve being expanded on in the manual, or even tutorial:

- Advanced keybinding configuration
  - [ ] Schemes and keymaps.
    - [x] Scheme inheritance (i.e. `:import` keyword of `define-scheme`).
  - [x] Rebinding/shadowing existing keybindings.
  - [ ] Binding nested keymaps.
  - [ ] Intercepting keyboard events with `command-dispatcher` and `input-skip-dispatcher`.
- Search engines
  - [x] Snippet to add Google as a (default) search engine?
  - [x] Explanation of `search-engine` class?
  - [ ] Search completion functions?
- [ ] Short explanation of differences with Emacs, so that it's clear we share no code with Emacs and Emacs configs won't work.
- [ ] Installation of WebExtensions.
- History tree
  - [ ] When the branching happens and how it is useful.
  - [ ] Which commands utilize branching and the tree-like structure in general?
- UI
  -  [ ] Prompt buffer usage and configuration.
  - [x] `lisp:` and `nyxt:` schemes.
  - [x] Internal pages.
  - [x] Internal schemes (#2049).
  - [ ] Font config, as in https://github.com/atlas-engineer/nyxt/issues/901#issuecomment-1202969223.
- How-Tos
  - Look up all the questions asked on Reddit and DIscourse.

EDIT: Add history and UI clusters.
UPD: Add nested keymaps point and update internal schemes point.
UPD: How-Tos

**Comment by Ambrevar** [comment]
Thanks for summing this up!

**Comment by guojing0** [comment]
And WebExtensions doc?

**Comment by aartaka** [comment]
> And WebExtensions doc?

Yes, but there's an issue for this, so I was hesitant about mentioning it there :)

Do we close the #1978 and expand the list here then?

**Comment by guojing0** [comment]
Sure.

**Comment by aartaka** [comment]
> Sure.

Thanks! I've added the point on WebExtension instalation :)

**Comment by guojing0** [comment]
For the Emacs part, maybe if the users reading [this](https://nyxt.atlas.engineer/article/why-building-nyxt-instead-of-an-emacs-package.org) and [this](https://nyxt.atlas.engineer/article/article-how-can-i-make-emacs-my-web-browser.org) should help clarify...

**Comment by aartaka** [comment]
Yet another point for improving the help system:
Thi ["Grand Unified Theory of Documentation"](https://documentation.divio.com/) [suggested on Discourse](https://discourse.atlas.engineer/t/nyxt-cookbook-tldr-configurations/420/4) suggests four groups of documentation:
- Tutorials: we have `tutorial`.
- Explanation: we have `tutorial` and `manual`.
- Reference: we have `describe-*` functions and `manual`.
- How-to guides: ?

It might be beneficial to just go through the questions on Reddit and write some snippets completing the things people most often ask for.

I feel like I've written about it somewhere, but I can't recall where...

**Comment by jmercouris** [comment]
Expanding the manual is not limited to 3.0.0, nor is it a blocker. This is a perpetual task that will never be *truly* finished!

**Comment by aadcg** [comment]
Too broad of an issue. Our manual is terrible but should only devote time to it when Nyxt is more mature.
